config/v1/types_infrastructure.go

```diff
@@ -853,6 +853,11 @@ type IBMCloudPlatformStatus struct {
 	// ResourceGroupName is the Resource Group for new IBMCloud resources created for the cluster.
 	ResourceGroupName string `json:"resourceGroupName,omitempty"`
 
+	// networkResourceGroupName is the Resource Group for network resources like the VPC and Subnets used by the cluster.
+	// If empty, the value is same as ResourceGroupName.
+	// +optional
+	NetworkResourceGroupName string `json:"networkResourceGroupName,omitempty"`
```

Comment:
How does this get set? Is the installer setting this and, is there any chance a user might want to set this day 2?

Reply:
The `installer` will eventually set this value, similar to the process used by Azure to manage BYON resources, once this change is merged, we can update the `installer`'s `api` version, and IBM Cloud MAPI's `api` version, so they can utilize this value.

At this time I don't expect a user to change this value, as I believe resources cannot be migrated to different ResourceGroups (I plan to confirm with IBM Cloud VPC on that). Changing it after cluster creation, with the expectation of using a different VPC/Subnets for worker creation could be possible, requiring some intricate networking configuration (between original VPC/Subnets from `installer` and a new VPC/Subnets in a different NetworkResourceGroup), but that sounds like a rare case, I would think.
Assuming migrating resources between ResourceGroups isn't supported on IBM Cloud.

We plan to add this support to the `installer` and MAPI in the 4.13 release, but need this change first obviously. But waiting until 4.13 changes are open would be acceptable I think. I can put this change on hold until 4.13 development is ready.

/hold

Reply:
4.13 branches on the 28th of October, lets pick this back up then

Reply:
If we aren't going to support this being mutated, we could add a validation that prevents it from being mutated.
You should also add to the godoc for the field that it may not be changed post installation

Reply:
Okay, let me see if I can work on adding those changes too

Reply:
Done, hopefully correctly.

Reply:
You'll need to make sure to test this validation, AFAIK the status is applied as an update so this might need to be changed to allow `empty -> populated` and then be immutable